backend/src/apiserver/server/run_server.go

```diff
@@ -30,6 +30,22 @@ type RunServer struct {
 
 func (s *RunServer) CreateRun(ctx context.Context, request *api.CreateRunRequest) (*api.RunDetail, error) {
 	err := s.validateCreateRunRequest(request)
+	if err != nil {
+		return nil, util.Wrap(err, "Validate create run request failed.")
+	}
```

Comment:
Can you instead create a generic function?

isAuthorized(resourceManager, context, resourcereferences){
  if !common.IsKubeflowDeployment() return true;  // avoid unecessary GetUserIdentity/GetNamespace call
  GetUserIdentity(context)
  GetNamespace(resourcereferences)
  return Authorize(...)  
}

I believe this is generic enough that will be reused for future APIs.

Reply:
SG. Done.